pkg/operator/controller/controller_lb.go

```diff
@@ -108,6 +106,20 @@ func (r *reconciler) currentLoadBalancerService(ci *ingressv1alpha1.ClusterIngre
 	return service, nil
 }
 
+// enforceLoadBalancerServiceFinalizer adds loadBalancerServiceFinalizer to service
+// if it doesn't exist.
+func (r *reconciler) enforceLoadBalancerServiceFinalizer(service *corev1.Service) error {
+	// TODO: Figure out why DNS updates apply DeletionTimestamp to loadbalancer service.
```

Comment:
I added this `TODO` because testing uncovered the following error. The error is attributed to the loadbalancer service being recreated after the service is assigned an `EXTERNAL-IP` and an DNS gets updated. I can see `DeletionTimestamp` is applied to the initial load balancer service.

```
2019-03-01T14:02:23.263-0800	ERROR	operator.init.kubebuilder.controller	controller/controller.go:217	Reconciler error	{"controller": "operator-controller", "request": "openshift-ingress-operator/default", "error": "failed to ensure clusteringress openshift-ingress-operator/default: failed to enforce load balancer service finalizer for ingress openshift-ingress-operator/default: Service \"router-default\" is invalid: metadata.finalizers: Forbidden: no new finalizers can be added if the object is being deleted, found new finalizers []string{\"ingress.openshift.io/load-balancer-service\"}", "errorCauses": [{"error": "failed to ensure clusteringress openshift-ingress-operator/default: failed to enforce load balancer service finalizer for ingress openshift-ingress-operator/default: Service \"router-default\" is invalid: metadata.finalizers: Forbidden: no new finalizers can be added if the object is being deleted, found new finalizers []string{\"ingress.openshift.io/load-balancer-service\"}"}]}
```